mlflow/server/js/src/gateway/components/endpoint-form/CreateEndpointModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { Modal } from '@databricks/design-system';
+import { useIntl } from 'react-intl';
+import { FormProvider } from 'react-hook-form';
+import { EndpointFormRenderer } from './EndpointFormRenderer';
+import { useCreateEndpointForm } from '../../hooks/useCreateEndpointForm';
+import { getReadableErrorMessage } from '../../utils/errorUtils';
+import type { Endpoint } from '../../types';
+
+interface CreateEndpointModalProps {
+  open: boolean;
+  onClose: () => void;
+  onSuccess?: (endpoint: Endpoint) => void;
+}
+
+export const CreateEndpointModal = ({ open, onClose, onSuccess }: CreateEndpointModalProps) => {
+  const intl = useIntl();
+
+  const {
+    form,
+    isLoading,
+    error,
+    resetErrors,
+    selectedModel,
+    isFormComplete,
+    handleSubmit,
+    handleCancel,
+    handleNameBlur,
+  } = useCreateEndpointForm({
+    onSuccess: (endpoint) => {
+      form.reset();
+      onSuccess?.(endpoint);
+      onClose();
+    },
+    onCancel: () => {
+      form.reset();
+      resetErrors();
+      onClose();
+    },
+  });
+
+  return (
+    <Modal
+      componentId="mlflow.gateway.create-endpoint-modal"
+      title={intl.formatMessage({
+        defaultMessage: 'Create endpoint',
+        description: 'Title for create endpoint modal',
+      })}
+      visible={open}
+      onCancel={handleCancel}
+      footer={null}
+      size="wide"
+      css={{ width: '100% !important' }}
```

Comment:
this may break in the near future as radix modals will get into OSS (but OTOH we'll get `extraWide` size by then), let's keep an eye on that